### DISCUSSION TITLE
Plots taking almost 1600 seconds each! How to speed up?

### DISCUSSION THREAD

**Discussion by greenham13** [madMAx43v3r/chia-plotter#313] [Q&A]
[P1] Table 1 took 5.02938 sec
[P1] Table 2 took 177.237 sec, found 4294981378 matches
[P1] Table 3 took 174.462 sec, found 4295056816 matches
[P1] Table 4 took 182.441 sec, found 4295137262 matches
[P1] Table 5 took 181.509 sec, found 4295213348 matches
[P1] Table 6 took 179.224 sec, found 4295168726 matches
[P1] Table 7 took 168.323 sec, found 4295084631 matches
Phase 1 took 1068.26 sec
[P2] max_table_size = 4295213348
[P2] Table 7 scan took 3.15198 sec
[P2] Table 7 rewrite took 38.5371 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 8.68713 sec
[P2] Table 6 rewrite took 18.469 sec, dropped 581434332 entries (13.5369 %)
[P2] Table 5 scan took 8.40049 sec
[P2] Table 5 rewrite took 17.7541 sec, dropped 762200133 entries (17.7453 %)
[P2] Table 4 scan took 8.27751 sec
[P2] Table 4 rewrite took 17.3893 sec, dropped 828984575 entries (19.3005 %)
[P2] Table 3 scan took 8.22354 sec
[P2] Table 3 rewrite took 17.3131 sec, dropped 855114869 entries (19.9093 %)
[P2] Table 2 scan took 8.18254 sec
[P2] Table 2 rewrite took 17.3735 sec, dropped 865548972 entries (20.1526 %)
Phase 2 took 192.875 sec
Wrote plot header with 268 bytes
[P3-1] Table 2 took 27.1467 sec, wrote 3429432406 right entries
[P3-2] Table 2 took 12.7156 sec, wrote 3429432406 left entries, 3429432406 final
[P3-1] Table 3 took 27.166 sec, wrote 3439941947 right entries
[P3-2] Table 3 took 14.2403 sec, wrote 3439941947 left entries, 3439941947 final
[P3-1] Table 4 took 27.188 sec, wrote 3466152687 right entries
[P3-2] Table 4 took 14.1271 sec, wrote 3466152687 left entries, 3466152687 final
[P3-1] Table 5 took 27.639 sec, wrote 3533013215 right entries
[P3-2] Table 5 took 15.1139 sec, wrote 3533013215 left entries, 3533013215 final
[P3-1] Table 6 took 28.8544 sec, wrote 3713734394 right entries
[P3-2] Table 6 took 14.7729 sec, wrote 3713734394 left entries, 3713734394 final
[P3-1] Table 7 took 32.0256 sec, wrote 4295084631 right entries
[P3-2] Table 7 took 17.391 sec, wrote 4294967296 left entries, 4294967296 final
Phase 3 took 265.185 sec, wrote 21877241945 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 42.9003 sec, final plot size is 108835902318 bytes
Total plot creation time was 1569.3 sec

I have lots of free ram - 600GB or so.  Any tips to try?

**Comment by kimobebe** [comment]
more threads and speed up your SSD

**Comment by kacperapi** [comment]
What is your hardware? This might help to find bottleneck

**Comment by therealflinchy** [comment]
your phase 2-3-4 are incredibly fast, let down by your P1. what's your CPU?

**Reply by greenham13** [reply]
3975 w/ 1GB RAM.  This one was 60 threads, both temp drives are ramdisks.

**Reply by digitalspaceport** [reply]
do some benching DD on the ram vs your raid arrays and see what they top out at. You should also check your numa bios settings and I would do 64 threads, not 60
